src/cli/plugin/pluginDownloader.js

```diff
@@ -1,39 +1,35 @@
-var _ = require('lodash');
-var zlib = require('zlib');
-var Promise = require('bluebird');
-var url = require('url');
-var fs = require('fs');
-var request = require('request');
-var tar = require('tar');
-var progressReporter = require('./progressReporter');
+let _ = require('lodash');
+let Promise = require('bluebird');
+let urlParse = require('url').parse;
+let fs = require('fs');
+let request = require('request');
+let progressReporter = require('./progressReporter');
 
 module.exports = function (settings, logger) {
+  let archiveType = undefined;
```

Comment:
`let archiveType;` would have it implicitly initialized with undefined as the value